xdiff --git a/src/main/java/test/TestService.java b/src/main/java/test/TestService.java
index cef524e..fab825e 100644
--- a/src/main/java/test/TestService.java
+++ b/src/main/java/test/TestService.java
@@ -109,4 +109,8 @@ public class TestService {
 		logger.debug("device info transform {} -> {}", deviceInfo, info);
 		return info;
 	}
+
+	public Mono<Integer> forTest() {
+		return Mono.just(100);
+	}
 }
diff --git a/src/test/java/test/TestServiceTest.java b/src/test/java/test/TestServiceTest.java
new file mode 100644
index 0000000..e0bbd9f
--- /dev/null
+++ b/src/test/java/test/TestServiceTest.java
@@ -0,0 +1,24 @@
+package test;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class TestServiceTest {
+
+
+	@Autowired
+	TokenService service;
+
+	@Test
+	public void test() {
+		assertEquals(100, service.forTest().block().intValue());
+	}
+
+}
\ No newline at end of file
